doc/dev/mountpoints.md

```diff
@@ -32,7 +32,13 @@ The _mountpoint config_ is defined by the keys below `system:/elektra/mountpoint
 It is merged into the configuration keyset of every plugin of this mountpoint (including the backend plugin).
 Specifically, the mountpoint config is put into the `system:/` namespace, while the config from `system:/elektra/mountpoints/<mountpoint>/plugins/<ref>/config` is put into the `user:/` namespace of the keyset passed to a plugin.
 
-<!-- TODO: cover config/needs dynamic config; put into default:/ at runtime ? -->
+<!-- TODO: implement `config/needs` -->
+
+Additionally, there is the `config/needs` part of a plugins contract.
+This part of the contract can be used by a plugin to provide configuration for another plugin it depends on.
+In particular, these keys are put into the `default:/` namespace, meaning this config can be overriden by both the _mountpoint config_ in `system:/elektra/mountpoints/<mountpoint>/config` and the plugin's config in `system:/elektra/mountpoints/<mountpoint>/plugins/<ref>/config`.
+
+<!-- TODO: swap defaul:/ and system:/ to mimic current behaviour? -->
```

Comment:
Currently, `config/needs` takes precedence over the mountpoint configuration provided to `kdb mount -c`. I'm not sure which way is better. On the one hand `config/needs` feels more like defaults to me, on the other hand, if the mountpoint configuration overrides it, the whole "load `config/needs` during `kdbOpen`" is pointless.

Reply:
Is it still the case that both configs are available and it is only about precedence? Imho it should be the other way round, when the user explicitly asked for some config, it should be preferred.

So the order of preference is (highest first):

- plugin specific config (kdb mount ... plugin `<config>`), could be in `dir:` namespace
- config passed by user (kdb mount `-c <config>` ...), could be in `user:` namespace
- config calculated from backend (`config/needs` and/or other config that might stem from backend), could be in `system:` namespace

@flo91 did you already touch that topic?

Reply:
> So the order of preference is (highest first):

True, we can just move the plugin config to `dir:/`. Then a plugin could add it's own `default:/` keys before calling `ksLookup (config, ...)` and have defaults always available.

> config calculated from backend (`config/needs` and/or other config that might stem from backend)

What other config?

Currently there is:

1. Plugin specific config (`kdb mount ... plugin <config>`) stored in `system:/elektra/mountpoints/<mountpoint>/plugins/<ref>/config`
2. Mountpoint config (`kdb mount -c <config> ...`) stored in `system:/elektra/mountpoints/<mountpoint>/config`
3. `config/needs` loaded from plugins contracts

I don't know of any other way to provide plugin config

Reply:
> What other config?

Maybe some backends will provide extensions to the contracts or want to append other configs. But yes, this is hypothetical, these 3 categories are perfect for now and cover everything we support.

Reply:
> Maybe some backends will provide extensions to the contracts or want to append other configs

That would need some changes, but should be doable (not for now though).

I'm thinking:

- Always load the backend plugin of a mountpoint last
- Add an API that let's plugins access the contract of other plugins
- Add an API that let's plugins get a list of other plugins on the same mountpoint
- Since the contract is a `KeySet*` created via function call anyway, the plugin could look at what other plugins are there and adapt it's contract (including `config/needs`) accordingly